src/OpenTelemetry/Metrics/Exemplar/ReadOnlyExemplarCollection.cs

```diff
@@ -0,0 +1,125 @@
+// Copyright The OpenTelemetry Authors
+// SPDX-License-Identifier: Apache-2.0
+
+using System.Diagnostics;
+#if EXPOSE_EXPERIMENTAL_FEATURES && NET8_0_OR_GREATER
+using System.Diagnostics.CodeAnalysis;
+using OpenTelemetry.Internal;
+#endif
+
+namespace OpenTelemetry.Metrics;
+
+#if EXPOSE_EXPERIMENTAL_FEATURES
+/// <summary>
+/// A read-only collection of <see cref="Exemplar" />s.
+/// </summary>
+/// <remarks><inheritdoc cref="Exemplar" path="/remarks"/></remarks>
+#if NET8_0_OR_GREATER
+[Experimental(DiagnosticDefinitions.ExemplarExperimentalApi, UrlFormat = DiagnosticDefinitions.ExperimentalApiUrlFormat)]
+#endif
+public
+#else
+internal
+#endif
+    readonly struct ReadOnlyExemplarCollection
+{
+    private readonly Exemplar[] exemplars;
+
+    internal ReadOnlyExemplarCollection(Exemplar[] exemplars)
+    {
+        Debug.Assert(exemplars != null, "exemplars was null");
+
+        this.exemplars = exemplars!;
+    }
+
+    /// <summary>
+    /// Gets the maximum number of <see cref="Exemplar" />s in the collection.
+    /// </summary>
+    /// <remarks>
+    /// Note: Enumerating the collection may return fewer results depending on
+    /// which <see cref="Exemplar"/>s in the collection received updates.
+    /// </remarks>
+    public int MaximumCount => this.exemplars.Length;
```

Comment:
I think we should mark this internal. This seems like an implementation detail that the user need not know.

Reply:
https://github.com/open-telemetry/opentelemetry-dotnet/pull/5386/commits/e719e5592a45b8de9020988c4ca5b8c948898141

I was thinking it might be useful for exporters which need to allocate some space before writing exemplars into them but I went ahead and made it `internal`.